src/Control/Eff/Writer/Strict.hs

```diff
@@ -90,3 +95,33 @@ runFirstWriter = runWriter (\w b -> Just w <|> b) Nothing
 -- | Handle Writer requests by overwriting previous values.
 runLastWriter :: Eff (Writer w ': r) a -> Eff r (a, Maybe w)
 runLastWriter = runWriter (\w b -> b <|> Just w) Nothing
+
+-- | Handle Writer requests, using a user-provided function to accumulate
+--   values and effect result.
+execWriter :: (w -> b -> b) -> b -> Eff (Writer w ': r) a -> Eff r b
+execWriter accum !b = fmap snd . runWriter accum b
+{-# INLINE execWriter #-}
+
+-- | Handle Writer requests, using a List to accumulate values and discard the
+--   effect result.
```

Comment:
@power-fungus minor nitpick, but could we make the documentation consistent with `execState` etc? e.g. focusing on how the monoidal value is handled (instead of the result of computation). something like  `Handle Writer requests, using a List to accumulate values and returning the final accumulated values`

if you agree this helps, could you tweak? if you don't, please ignore this

Reply:
good point. I did not think about the wording.